apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/WriteTests.cs

```diff
@@ -0,0 +1,190 @@
+                    new SpannerParameter("Float64Value", SpannerDbType.Float64, 2.0),
+                    new SpannerParameter("StringValue", SpannerDbType.String, "abc"),
+                    new SpannerParameter("TimestampValue", SpannerDbType.Timestamp, testTimestamp),
+                    new SpannerParameter("DateValue", SpannerDbType.Date, testDate),
+                    new SpannerParameter("BoolArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Bool),
+                        bArray),
+                    new SpannerParameter("Int64ArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Int64),
+                        lArray),
+                    new SpannerParameter("Float64ArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Float64),
+                        dArray),
+                    new SpannerParameter("StringArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.String),
+                        sArray),
+                    new SpannerParameter("BytesArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Bytes),
+                        bArrayArray),
+                    new SpannerParameter("TimestampArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Timestamp),
+                        tmArray),
+                    new SpannerParameter("DateArrayValue",
+                        SpannerDbType.ArrayOf(SpannerDbType.Date),
+                        dtArray)
+                }));
+            using (var reader = await GetLastRowAsync())
+            {
+                Assert.True(reader.GetFieldValue<bool>(reader.GetOrdinal("BoolValue")));
+                Assert.Equal(1, reader.GetFieldValue<long>(reader.GetOrdinal("Int64Value")));
+                Assert.True(
+                    Math.Abs(2.0 - reader.GetFieldValue<double>(reader.GetOrdinal("Float64Value")))
+                    < double.Epsilon);
+                Assert.Equal("abc", reader.GetFieldValue<string>(reader.GetOrdinal("StringValue")));
+                Assert.Equal(testTimestamp, reader.GetFieldValue<DateTime>(reader.GetOrdinal("TimestampValue")));
+                Assert.Equal(testDate, reader.GetFieldValue<DateTime>(reader.GetOrdinal("DateValue")));
+                Assert.Equal(bArray,
+                    reader.GetFieldValue<bool?[]>(reader.GetOrdinal("BoolArrayValue")));
+                Assert.Equal(lArray,
+                    reader.GetFieldValue<long?[]>(reader.GetOrdinal("Int64ArrayValue")));
+                Assert.Equal(dArray,
+                    reader.GetFieldValue<double?[]>(reader.GetOrdinal("Float64ArrayValue")));
+                Assert.Equal(sArray,
+                    reader.GetFieldValue<string[]>(reader.GetOrdinal("StringArrayValue")));
+                Assert.Equal(bArrayArray,
+                    reader.GetFieldValue<string[]>(reader.GetOrdinal("BytesArrayValue")));
```

Comment:
yes -- but there are still a whole range of type conversion possibilities that I have not covered that need to be covered in unit tests.  I'll get 100% type conversion coverage when I add those.